CodeForBaltimore/Bmore-Responsive
# 207: Add bearer token support

# Description

- Added bearer token parsing support
- Deprecated 'token' header usage
- Updated initial set of tests to use new header
- Updated documentation

resolves #207 

## Related PRs

- N/A

## Todos


## Deploy Notes

- N/A

## Steps to Test or Reproduce

1. Verify that all tests pass
1. Verify that authentication works with both the `token` header and the `Authorization` header using the token with the `Bearer` scheme.
1. Verify that the OpenAPI contract shows all endpoints other than `/user/login` and `/health` as requiring authorization

## Impacted Areas in Application

List general components of the application that this PR will affect:

- Authentication
